.github/workflows/release.yaml

```diff
@@ -200,6 +200,48 @@ jobs:
       env:
         GH_TOKEN: ${{ secrets.REPO_ACCESS_TOKEN }}
 
+  build_push_docker_image:
+    needs: [build, release]
+    name: Build and push Docker image
+    permissions:
+      contents: read
+      packages: write # To push the Docker image to ghcr.io
+    uses: ./.github/workflows/_build_docker.yaml
+    with:
+      release-tag: ${{ needs.release.outputs.release-tag }}
+
+  publish_docker_imabe_sbom:
+    name: Release the Docker image sbom
+    needs: [build_push_docker_image, release]
+    permissions:
+      contents: write # To publish the SBOM of the Docker image.
+    runs-on: ubuntu-latest
+    env:
+      SYFT_BIN: ${{ github.workspace }}/bin
+      SBOM_FILE_NAME: macaron-sbom-cyclonedx.json
+    steps:
+    - name: Check out repository
```

Comment:
Why do we need to checkout the repository here?

Reply:
I believe we need to checkout the repository following what we already have for publishing the provenance with `gh relase upload` - https://github.com/oracle-samples/macaron/blob/9e89477f6ad91749bbb9bda1d7361ee0d1cb3305/.github/workflows/release.yaml#L180

Reply:
> I believe we need to checkout the repository following what we already have for publishing the provenance with `gh relase upload` -
> 
> https://github.com/oracle-samples/macaron/blob/9e89477f6ad91749bbb9bda1d7361ee0d1cb3305/.github/workflows/release.yaml#L180

Yes, that's right. `gh` needs a checked out git repo.